sound/soc/sof/intel/hda.c

```diff
@@ -1109,7 +1109,8 @@ static int hda_sdw_machine_select(struct snd_sof_dev *sdev)
 				break;
```

Comment:
@RanderWang For commit message:
Info.count indicates the number of hardware links. If  a platform does not
have all links connected, link_slaves_found() gets called with an invalid link object.
Fix this by limiting the loop in hda_sdw_machine_select() to valid links.

And maybe have this in summary as well to indicate this is fixing a problem, not just refining the logic. Something like:
ASoC: SOF: Intel: hda: ignore unused links in sdw machine select

Reply:
thanks, @kv2019i . you are pretty good at writing summary.